pkg/scaling/resolver/scale_resolvers.go

```diff
@@ -193,8 +194,13 @@ func ResolveAuthRefAndPodIdentity(ctx context.Context, client client.Client, log
 					fmt.Errorf("error getting service account: '%s', error: %w", serviceAccountName, err)
 			}
 			authParams["awsRoleArn"] = serviceAccount.Annotations[kedav1alpha1.PodIdentityAnnotationEKS]
-		} else if podIdentity.Provider == kedav1alpha1.PodIdentityProviderAwsKiam {
+		case kedav1alpha1.PodIdentityProviderAwsKiam:
 			authParams["awsRoleArn"] = podTemplateSpec.ObjectMeta.Annotations[kedav1alpha1.PodIdentityAnnotationKiam]
+		case kedav1alpha1.PodIdentityProviderAzure:
+			if podIdentity.GetIdentityID() == "" {
+				logger.Info("IdentityID of PodIdentity is empty.")
+			}
```

Comment:
As webhooks aren't mandatory and users could not install them, I still would check if it's `nil` before the value.l Otherwise a user who hasn't installed the webhooks will see a warning that shouldn't be there

Reply:
If considering that webhooks aren't mandatory, I think this checking should return err if value is empty.